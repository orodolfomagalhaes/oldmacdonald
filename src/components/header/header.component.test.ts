import { expect, it, afterEach, beforeEach, describe, vi } from 'vitest';
import {HeaderComponent} from "./header.component.ts";

describe("HeaderComponent", () => {
  let component: HTMLElement;

  beforeEach(() => {
    component = new HeaderComponent();
    document.body.appendChild(component);
  })

  afterEach(() => {
    document.body.innerHTML = ``;
  })

  it('should create the search component', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('should dispatch add-new-farm event when button is clicked', () => {
    const button: any = document.querySelector('#add-farm');
    const eventListener = vi.fn();
    document.addEventListener('add-new-farm', eventListener);
    button?.click();
    expect(eventListener).toHaveBeenCalled();
    document.removeEventListener('add-new-farm', eventListener);
  });
})