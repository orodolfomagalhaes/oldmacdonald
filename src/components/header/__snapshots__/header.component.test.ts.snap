// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderComponent > should create the search component 1`] = `
<header-component>
  <div
    class="header d-flex justify-content-between align-items-end"
  >
          
      
    <div
      class="company-logo"
    >
      
        
      <h1>
        Farms
      </h1>
      
        
      <span
        class="text-body-secondary"
      >
        Manage your farms.
      </span>
      
      
    </div>
    
      
    <div
      class="actions"
    >
      
        
      <button
        aria-label="Add a new farm"
        class="btn btn-primary"
        id="add-farm"
        type="button"
      >
        
            
        <i
          aria-hidden="true"
          class="fa fa-plus me-1"
        />
        
            Add Farm
        
      </button>
      
      
    </div>
                  
    
  </div>
     
    
  <hr />
</header-component>
`;
